pkg/operators/v1/operatorgroup_types.go

```diff
@@ -45,6 +45,49 @@ type OperatorGroupSpec struct {
 	// Static tells OLM not to update the OperatorGroup's providedAPIs annotation
 	// +optional
 	StaticProvidedAPIs bool `json:"staticProvidedAPIs,omitempty"`
+
+	UpgradeStrategy *UpgradeStrategy `json:"upgradeStrategy,omitempty"`
+}
+
+// UpgradeStrategy defines the upgrade strategy for operators in the namespace.
+// There are currently two supported upgrade strategies:
+//
+// Default: OLM will only allow clusterServiceVersions to move to the replacing
+// phase from the succeeded phase. This effectively means that OLM will not
+// allow operators to move to the next version if an installation or upgrade
+// has failed.
+//
+// UnsafeFailForward: OLM will allow clusterServiceVersions to move to the
+// replacing phase from the succeeded phase or from the failed phase.
+// Additionally, OLM will generate new installPlans when a subscription references
+// a failed installPlan and the catalog has been updated with a new upgrade for
+// the existing set of operators. This upgrade strategy is unsafe and may result
+// in unexpected behavior or unrecoverable data loss unless you have deep
+// understanding of the set of operators being managed in the namespace.
+type UpgradeStrategy struct {
+	// +kubebuilder:validation:Enum=Default;UnsafeFailForward
+	// +kubebuilder:default=Default
+	Name upgradeStrategyName `json:"name"`
+}
+
+type upgradeStrategyName string
+
+const (
+	DefaultUpgradeStrategy           upgradeStrategyName = "Default"
+	UnsafeFailForwardUpgradeStrategy upgradeStrategyName = "UnsafeFailForward"
+)
+
+func (o *OperatorGroup) UpgradeStrategy() upgradeStrategyName {
+	if o.Spec.UpgradeStrategy == nil {
+		return DefaultUpgradeStrategy
+	}
+
+	switch {
+	case o.Spec.UpgradeStrategy.Name == UnsafeFailForwardUpgradeStrategy:
+		return o.Spec.UpgradeStrategy.Name
+	default:
+		return DefaultUpgradeStrategy
+	}
 }
```

Comment:
Based on my above suggestion about making `UpdateStrategy` a non-pointer, I wonder if this function can just be dropped given the defaulting and validation we include in the CRD?

Reply:
Sounds reasonable, I only added this method to handle the `nil UpgradeStrategy` case.

Reply:
Unfortunately, due to the reasoning described [here](https://github.com/operator-framework/api/pull/235#discussion_r843161019), we can't rely on the CRD defaulting to set the correct values.